swift/Ice/helloUI/Client.swift

```diff
@@ -0,0 +1,208 @@
+            return
+        }
+        firstly {
+            helloPrx.ice_flushBatchRequestsAsync()
+        }.done {
+            self.flushBatchSend()
+        }.catch { error in
+            self.exception(error)
+        }
+    }
+
+    func shutdown() {
+        do {
+            if helloPrx == nil {
+                return
+            }
+
+            if proxySettings.deliveryMode != .OnewayBatch,
+               proxySettings.deliveryMode != .OnewaySecureBatch,
+               proxySettings.deliveryMode != .DatagramBatch
+            {
+                sendingRequest()
+                var response = false
+                firstly {
+                    helloPrx.shutdownAsync { _ in
+                        if !response {
+                            self.requestSent()
+                        }
+                    }
+                }.done {
+                    response = true
+                    self.ready()
+                }.catch { error in
+                    response = true
+                    self.exception(error)
+                    self.ready()
+                }
+            } else {
+                try helloPrx.shutdown()
+                queuedRequest("shutdown")
+            }
+        } catch {
+            exception(error)
+            ready()
+        }
+    }
+
+    func updateProxy() throws {
+        let mode = proxySettings.deliveryMode
+
+        if proxySettings.hostname == "" {
+            helloPrx = nil
+            return
+        }
+
+        let s = "hello:tcp -h \"\(proxySettings.hostname)\" -p 10000:" +
+            "ssl -h \"\(proxySettings.hostname)\" -p 10001:" +
+            "udp -h \"\(proxySettings.hostname)\" -p 10000"
+
+        UserDefaults.standard.set(proxySettings.hostname, forKey: hostnameKey)
+        helloPrx = uncheckedCast(prx: try communicator.stringToProxy(s)!, type: HelloPrx.self)
+
+        switch mode {
+        case .Twoway:
+            helloPrx = helloPrx.ice_twoway()
+        case .TwowaySecure:
+            helloPrx = helloPrx.ice_twoway().ice_secure(true)
+        case .ModeOneway:
+            helloPrx = helloPrx.ice_oneway()
+        case .OnewayBatch:
+            helloPrx = helloPrx.ice_batchOneway()
+        case .OnewaySecure:
+            helloPrx = helloPrx.ice_oneway().ice_secure(true)
+        case .OnewaySecureBatch:
+            helloPrx = helloPrx.ice_batchOneway().ice_secure(true)
+        case .Datagram:
+            helloPrx = helloPrx.ice_datagram()
+        case .DatagramBatch:
+            helloPrx = helloPrx.ice_batchDatagram()
+        }
+
+        if proxySettings.timeout != 0 {
+            helloPrx = helloPrx.ice_invocationTimeout(Int32(proxySettings.timeout))
+        }
+    }
+
+    func sendingRequest() {
+        helloEnabled = false
+        shutdownEnabled = false
+        statusMessage = "Sending request"
+        isSpinning = true
+    }
+
+    func requestSent() {
+            if let connection = helloPrx.ice_getCachedConnection() {
```

Comment:
I think this hostname overwriting doesn't make sense for helloUI: you've entered hostname, obviously that's the hostname the Client is going to use to connect to the server. Here we're just replacing hostname with the remote IP address of the connection established with the server.